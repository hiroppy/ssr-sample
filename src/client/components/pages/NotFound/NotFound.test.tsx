import * as React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as sinon from 'sinon';
import { MemoryRouter } from 'react-router-dom';
import { NotFound } from '.';

test('should render self and sub-components', () => {
  const tree = mount(
    <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
      <NotFound />
    </MemoryRouter>
  );

  expect(toJson(tree)).toMatchSnapshot();
});

test('should call stopSaga', () => {
  const stop = sinon.spy();

  shallow(<NotFound stopSaga={stop} />);

  expect(stop.calledOnce).toBeTruthy();
});
