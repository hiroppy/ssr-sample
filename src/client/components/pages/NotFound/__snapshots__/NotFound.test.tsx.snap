// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render self and sub-components 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 15px 30px;
}

.c0 > *:not(:first-child) {
  margin-top: 24px;
}

<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NotFound>
      <Component
        title="404"
      >
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <SideEffect(NullComponent)
            defer={true}
            encodeSpecialCharacters={true}
            link={
              Array [
                Object {
                  "href": "https://fonts.googleapis.com/css?family=Muli",
                  "rel": "stylesheet",
                },
              ]
            }
            meta={
              Array [
                Object {
                  "charset": "UTF-8",
                },
                Object {
                  "content": "width=device-width, initial-scale=1.0",
                  "name": "viewport",
                },
              ]
            }
            title="404"
            titleAttributes={Object {}}
          >
            <NullComponent
              defer={true}
              encodeSpecialCharacters={true}
              link={
                Array [
                  Object {
                    "href": "https://fonts.googleapis.com/css?family=Muli",
                    "rel": "stylesheet",
                  },
                ]
              }
              meta={
                Array [
                  Object {
                    "charset": "UTF-8",
                  },
                  Object {
                    "content": "width=device-width, initial-scale=1.0",
                    "name": "viewport",
                  },
                ]
              }
              title="404"
              titleAttributes={Object {}}
            />
          </SideEffect(NullComponent)>
        </HelmetWrapper>
      </Component>
      <Component>
        <styled.div>
          <div
            className="c0"
          >
            <p>
              Page not found
            </p>
            <Link
              replace={false}
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Go Back to Top Page
              </a>
            </Link>
          </div>
        </styled.div>
      </Component>
    </NotFound>
  </Router>
</MemoryRouter>
`;
